ui/desktop/app/routes/application.js

```diff
@@ -34,9 +33,13 @@ export default class ApplicationRoute extends Route {
   async beforeModel() {
     this.intl.setLocale(['en-us']);
     await this.session.setup();
-    const theme = this.session.get('data.theme');
-    this.toggleTheme(theme);
     await this.clusterUrl.updateClusterUrl();
+    const theme = this.session.get('data.theme');
+    /* eslint-disable-next-line ember/no-controller-access-in-routes */
+    const controller = this.controllerFor(
```

Comment:
Hmm I don't think we should reach down into a child controller. I'm thinking the `toggleTheme` action you have in the controller should either be in a service or in this `application` controller and then you call it from the settings controller

Reply:
thanks for the feedback, I moved it to app controller